### PR TITLE
OpsGenie Integrations

### PR DESCRIPTION
**Pull request by gshriki**
OpsGenie new integration to allow getting on call users from specific schedules

### CODE REVIEW

File: Integrations/integration-OpsGenie.yml [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+detaileddescription: OpsGenie integration allows querying for the current on-call
+  users for a given Schedule. The integration requires an API key, that can
+  be acquired from the OpsGenie interface, under Integrations / API.
+configuration:
+- display: API Key
+  name: apiKey
+  defaultvalue: ""
+  type: 4
+  required: true
+- display: User system proxy configuration
```
Comment (Spivaka) [inline Integrations/integration-OpsGenie.yml]: insecure?
+ usually the order is
server url
api key
proxy and insecure

File: Integrations/integration-OpsGenie.yml [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+            params.proxy
+        );
+
+        if (res.StatusCode < 200 || res.StatusCode >= 300) {
+            logError('OpsGenie error: Request failed for ' + requestUrl + '. Returned body is ' + JSON.stringify(res));
+            throw 'OpsGenie: Request Failed.\nStatus code: ' + res.StatusCode + '.\nBody: ' + JSON.stringify(res) + '.';
+        }
+
+        var jsonBody = '';
+        try {
```
Comment (Spivaka) [inline Integrations/integration-OpsGenie.yml]: better to just return inside the try

File: Integrations/integration-OpsGenie.yml [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+            sendRequest(url);
+            return 'ok';
+
+        case 'opsgenie-get-on-call':
+            return getOnCall(args.schedule);
+
+        case 'opsgenie-get-user':
+            return getUser(args.userID);
+
+        default:
```
Comment (Spivaka) [inline Integrations/integration-OpsGenie.yml]: can be removed, can't happen

File: Integrations/integration-OpsGenie.yml [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+        baseURL += '/';
+    }
+
+    var serviceURLs = {
+        schedules: 'schedules',
+        account: 'account',
+        users: 'users'
+    };
+
+    var sendRequest = function(url, param) {
```
Comment (Spivaka) [inline Integrations/integration-OpsGenie.yml]: you can remove params here as its not passed inside the call but a global

File: Integrations/integration-OpsGenie.yml (hunk 1 above)
Comment (gshriki) [reply Integrations/integration-OpsGenie.yml]: it's a cloud service, so no need insecure. will change order.

File: Integrations/integration-OpsGenie.yml (hunk 3 above)
Comment (gshriki) [reply Integrations/integration-OpsGenie.yml]: I think it's better to keep so if someone adds a command without adding the case, it's easier to see.

File: Integrations/integration-OpsGenie.yml (hunk 4 above)
Comment (gshriki) [reply Integrations/integration-OpsGenie.yml]: This is `param` and not `params`. it's a way to pass parameters to the API. 
But I'm not using it with the current APIs implemented, so removed,.

File: Integrations/integration-OpsGenie.yml (hunk 2 above)
Comment (gshriki) [reply Integrations/integration-OpsGenie.yml]: done

**Comment by gshriki** [comment]
@Spivaka - ping.

**Pull request merged**